boards/nxp/frdm_mcxn947/board.c

```diff
@@ -177,22 +190,27 @@ void board_early_init_hook(void)
 
 #if DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(gpio0))
 	CLOCK_EnableClock(kCLOCK_Gpio0);
+	unsecure_gpio(GPIO0);
 #endif
 
 #if DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(gpio1))
 	CLOCK_EnableClock(kCLOCK_Gpio1);
+	unsecure_gpio(GPIO1);
 #endif
 
 #if DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(gpio2))
 	CLOCK_EnableClock(kCLOCK_Gpio2);
+	unsecure_gpio(GPIO2);
 #endif
 
 #if DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(gpio3))
 	CLOCK_EnableClock(kCLOCK_Gpio3);
+	unsecure_gpio(GPIO3);
 #endif
 
 #if DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(gpio4))
 	CLOCK_EnableClock(kCLOCK_Gpio4);
+	unsecure_gpio(GPIO4);
```

Comment:
calling this routine for every instance makes me think it should go in a driver init routine

Reply:
that is another option, but not one I plan to spend time on right now.

Reply:
I'm the one who originally reported this issue (on Discord).
If you would like, I can help implement the change into the driver. What would be the best way to do it? Two functions with a `mask` parameter for pin control and two for interrupt control, and adapt the API accordingly?

Reply:
it seems like the code in the function being added here can just be added to the driver init routine

Reply:
Oh yes if we want the pins to be configurable for non-secure/non-privilege access all the time.
But the best would be to make it dynamically customizable, I guess.